src/package/abi/Microsoft.Windows.AbiWinRT.targets

```diff
@@ -6,8 +6,500 @@ Copyright (C) Microsoft Corporation. All rights reserved.
+            </_AbiWinRTCompInputs>
+            <_AbiWinRTCompRefs Remove="@(_AbiWinRTCompRefs)"/>
+            <_AbiWinRTCompRefs Include="@(AbiWinRTDirectWinMDReferences)"/>
+            <_AbiWinRTCompRefs Include="@(AbiWinRTDynamicProjectWinMDReferences)"/>
+            <_AbiWinRTCompRefs Include="@(AbiWinRTPlatformWinMDReferences)"/>
+        </ItemGroup>
+        <PropertyGroup>
+            <_AbiWinRTParameters>$(AbiWinRTCommandVerbosity) $(AbiWinRTParameters) $(AbiWinRTCommandEnsureSDKHeaderCompat) $(AbiWinRTCommandAddAbiNamespacePrefix)</_AbiWinRTParameters>
+            <_AbiWinRTParameters>$(_AbiWinRTParameters) @(_AbiWinRTCompInputs->'-in &quot;%(WinMDPath)&quot;', '&#x0d;&#x0a;')</_AbiWinRTParameters>
+            <_AbiWinRTParameters>$(_AbiWinRTParameters) @(_AbiWinRTCompRefs->'-ref &quot;%(WinMDPath)&quot;', '&#x0d;&#x0a;')</_AbiWinRTParameters>
+            <_AbiWinRTParameters>$(_AbiWinRTParameters) -out &quot;$(GeneratedFilesDir).&quot;</_AbiWinRTParameters>
+        </PropertyGroup>
+        <!-- Always write the AbiWinRT_comp.rsp file when the target runs, because the file is used as the output of this target. -->
+        <WriteLinesToFile
+            File="$(IntDir)AbiWinRT_comp.rsp" Lines="$(_AbiWinRTParameters)"
+            ContinueOnError="true" Overwrite="true" />
+        <Message Text="$(AbiWinRTCommand)" Importance="$(AbiWinRTVerbosity)" Condition="'@(_AbiWinRTCompInputs)' != ''"/>
+        <Exec Command="$(AbiWinRTCommand)" Condition="'@(_AbiWinRTCompInputs)' != ''"/>
+    </Target>
+
+    <Target Name="AbiWinRTMakeProjections" DependsOnTargets="AbiWinRTResolveReferences;AbiWinRTMakePlatformProjection;AbiWinRTMakeReferenceProjection;AbiWinRTMakeComponentProjection;$(AbiWinRTMakeProjectionsDependsOn)" />
+
+    <!--Append any additional item metadata after all default and project settings have been applied-->
+    <ItemDefinitionGroup>
+      <ClCompile>
+        <AdditionalOptions>%(AdditionalOptions) /bigobj /await</AdditionalOptions>
+        <AdditionalIncludeDirectories>%(AdditionalIncludeDirectories);$(GeneratedFilesDir)</AdditionalIncludeDirectories>
+      </ClCompile>
+      <Midl Condition="'$(AbiWinRTModernIDL)' != 'false'">
+        <AdditionalMetadataDirectories Condition="'%(AdditionalMetadataDirectories)' == '' And '$(WindowsSDK_MetadataFoundationPath)' != ''">$(WindowsSDK_MetadataFoundationPath);%(AdditionalMetadataDirectories)</AdditionalMetadataDirectories>
+        <AdditionalMetadataDirectories Condition="'%(AdditionalMetadataDirectories)' == '' And '$(WindowsSDK_MetadataFoundationPath)' == ''">$(WindowsSDK_MetadataPath);%(AdditionalMetadataDirectories)</AdditionalMetadataDirectories>
+        <AdditionalOptions>%(AdditionalOptions) /nomidl</AdditionalOptions>
+      </Midl>
+      <Link>
+        <AdditionalDependencies Condition="'$(AbiWinRTLibs)' != 'false'">%(AdditionalDependencies);WindowsApp.lib</AdditionalDependencies>
+      </Link>
+    </ItemDefinitionGroup>
+  
```

Comment:
remove all this stuff that's cppwinrt carryover

Reply:
Sure. For my own curiosity, why was this here? I wasn't 100% sure what `CppWinRTLibs` was for, is it for desktop projects that also want to link against WindowsApp.lib to access WinRT APIs?

Reply:
I'd say if you're not sure what it is, remove it.  You probably only need ~100 lines in this file.  I don't think you need to be configuring other tools like cl, midlrt, mdmerge, etc.  And you don't need all the platform/component projection targeting.

---
In reply to: [443886013](https://github.com/microsoft/xlang/pull/659#discussion_r443886013) [](ancestors = 443886013)